hedera-node/hedera-app/src/main/java/com/hedera/node/app/fees/ExchangeRateManager.java

```diff
@@ -121,16 +136,30 @@ public void update(@NonNull final Bytes bytes) {
         }
 
         // Check bounds
-        // TODO: This check needs to be skipped if payer is SysAdmin or Treasury
         final var ratesConfig = configProvider.getConfiguration().getConfigData(RatesConfig.class);
-        final var limitPercent = ratesConfig.intradayChangeLimitPercent();
-        if (!isNormalIntradayChange(midnightRates, proposedRates, limitPercent)) {
-            throw new HandleException(ResponseCodeEnum.EXCHANGE_RATE_CHANGE_LIMIT_EXCEEDED);
+        final var accountsConfig = configProvider.getConfiguration().getConfigData(AccountsConfig.class);
+        final var isSuperUser = isSuperUser(payerId, accountsConfig);
+
+        if (!isSuperUser) {
+            final var limitPercent = ratesConfig.intradayChangeLimitPercent();
+            if (!isNormalIntradayChange(midnightRates, proposedRates, limitPercent)) {
+                throw new HandleException(ResponseCodeEnum.EXCHANGE_RATE_CHANGE_LIMIT_EXCEEDED);
+            }
         }
 
         // Update the current ExchangeRateInfo and eventually the midnightRates
         this.currentExchangeRateInfo = new ExchangeRateInfoImpl(proposedRates);
-        // TODO: If payer is SysAdmin also update the midnightRates
+        // TODO: save the mignightRates in state only
+        if (isSuperUser) {
```

Comment:
This is not what is currently implemented in Mono. In Mono, only the SysAdmin can override the `midnightRates`.

Reply:
@tinker-michaelj thoughts?

Reply:
fixed it